### PR TITLE
Make bolt agent an object in integration tests

### PR DESCRIPTION
**Pull request by ConorNeo**

### CODE REVIEW

**Review by thelonelyvulpes — APPROVED** [review]
🍰

File: packages/neo4j-driver/test/examples.test.js [hunk 1]
```diff
@@ -828,8 +828,8 @@ describe('#integration examples', () => {
 
     const people = await result.toPromise()
     expect(people).toEqual([
-      Notification.createNext('Infinity Gauntlet'),
       Notification.createNext('Mjölnir'),
+      Notification.createNext('Infinity Gauntlet'),
```
Comment (bigmontz) [inline packages/neo4j-driver/test/examples.test.js]: This change won't solve the issue and also make it more easy to happen. We need a equal comparison which not depends on the order, actually. But for now, we can revert this change.

**Review by bigmontz — CHANGES_REQUESTED** [review]
We need revert the change on the examples test.

**Review by bigmontz — APPROVED** [review]
📦

File: packages/neo4j-driver/test/examples.test.js [hunk 2]
```diff
@@ -805,8 +805,8 @@ describe('#integration examples', () => {
     const tempSession = driver.session()
     try {
       await tempSession.run(
-        "UNWIND ['Infinity Gauntlet', 'Mjölnir'] AS item " +
-          'CREATE (:Product {id: 0, title: item})'
+        "UNWIND [{ id: 0, title: 'Infinity Gauntlet'} , { id: 1, title: 'Mjölnir' }] AS item " +
+        "CREATE (:Product {id: item.id, title: item.title})'"
```
Comment (bigmontz) [inline packages/neo4j-driver/test/examples.test.js]: Revert it

File: packages/neo4j-driver/test/examples.test.js [hunk 3]
```diff
@@ -827,11 +827,9 @@ describe('#integration examples', () => {
...
 
     const people = await result.toPromise()
-    expect(people).toEqual([
-      Notification.createNext('Infinity Gauntlet'),
-      Notification.createNext('Mjölnir'),
-      Notification.createComplete()
-    ])
+    expect(people).toContain(
+      Notification.createNext('Infinity Gauntlet')
+    )
```
Comment (bigmontz) [inline packages/neo4j-driver/test/examples.test.js:833]:
```suggestion
    expect(titles.length).toEqual(3)
    expect(people).toContain(
      Notification.createNext('Infinity Gauntlet')
    )
    expect(people).toContain(
      Notification.createNext('Mjölnir')
    )
    expect(people).toContain(
      Notification.createComplete()
    )
```

**Review by bigmontz — COMMENTED** [review]
I might solve the issue with this test.

File: packages/neo4j-driver/test/examples.test.js (hunk 2 above)
Comment (bigmontz) [inline packages/neo4j-driver/test/examples.test.js]:
```suggestion
        "UNWIND ['Infinity Gauntlet', 'Mjölnir'] AS item " +
        'CREATE (:Product {id: 0, title: item})'
```

File: packages/neo4j-driver/test/examples.test.js [hunk 4]
```diff
@@ -827,11 +827,16 @@ describe('#integration examples', () => {
     // end::rx-transaction-function[]
 
     const people = await result.toPromise()
-    expect(people).toEqual([
-      Notification.createNext('Infinity Gauntlet'),
-      Notification.createNext('Mjölnir'),
+    expect(titles.length).toEqual(3)
```
Comment (ConorNeo) [inline packages/neo4j-driver/test/examples.test.js]: do we mean people.length

**Pull request merged**